be/src/formats/parquet/group_reader.cpp

```diff
@@ -355,37 +346,57 @@ void GroupReader::_process_columns_and_conjunct_ctxs() {
     int read_col_idx = 0;
 
     for (auto& column : _param.read_cols) {
-        int chunk_index = column.col_idx_in_chunk;
         SlotId slot_id = column.slot_id;
-        const auto* parquet_field =
-                _param.file_metadata->schema().get_stored_column_by_field_idx(column.field_idx_in_parquet);
-        DCHECK(parquet_field != nullptr);
-        const tparquet::ColumnMetaData& column_metadata =
-                _row_group_metadata->columns[parquet_field->physical_column_index].meta_data;
-        if (_can_use_as_dict_filter_column(slots[chunk_index], conjunct_ctxs_by_slot, column_metadata)) {
-            _dict_filter_ctx.use_as_dict_filter_column(read_col_idx, slot_id, conjunct_ctxs_by_slot.at(slot_id));
-            _active_column_indices.emplace_back(read_col_idx);
-        } else {
-            bool has_conjunct = false;
-            if (conjunct_ctxs_by_slot.find(slot_id) != conjunct_ctxs_by_slot.end()) {
-                for (ExprContext* ctx : conjunct_ctxs_by_slot.at(slot_id)) {
+        if (conjunct_ctxs_by_slot.find(slot_id) != conjunct_ctxs_by_slot.end()) {
+            for (ExprContext* ctx : conjunct_ctxs_by_slot.at(slot_id)) {
+                std::vector<std::string> sub_field_path;
+                if (_try_to_use_dict_filter(column, ctx, sub_field_path,
+                                            slots[column.col_idx_in_chunk]->is_output_column())) {
+                    _use_as_dict_filter_column(read_col_idx, slot_id, sub_field_path);
+                } else {
                     _left_conjunct_ctxs.emplace_back(ctx);
                 }
-                has_conjunct = true;
             }
-            if (config::parquet_late_materialization_enable && !has_conjunct) {
+            _active_column_indices.emplace_back(read_col_idx);
+        } else {
+            if (config::parquet_late_materialization_enable) {
                 _lazy_column_indices.emplace_back(read_col_idx);
             } else {
                 _active_column_indices.emplace_back(read_col_idx);
             }
         }
         ++read_col_idx;
     }
+
     if (_active_column_indices.empty()) {
         _active_column_indices.swap(_lazy_column_indices);
     }
 }
 
+bool GroupReader::_try_to_use_dict_filter(const GroupReaderParam::Column& column, ExprContext* ctx,
+                                          std::vector<std::string>& sub_field_path, bool is_output_column) {
+    const Expr* root_expr = ctx->root();
+    std::vector<std::vector<std::string>> subfields;
+    root_expr->get_subfields(&subfields);
+
+    for (int i = 1; i < subfields.size(); i++) {
+        if (subfields[i] != subfields[0]) {
```

Comment:
What does this expression mean? Why check subfields equality?

Reply:
check if the expr contains more than one subfields, such as a.b.c + a.b.d > 10; there is one slot but two subfileds

Reply:
there also may be an expr contains more than one subfields but they are the same, like a.b.c > 10 or a.b.c < 5